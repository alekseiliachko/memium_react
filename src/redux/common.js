export const LOADING_STATUS = {
  NOT_LOADED: "NOT_LOADED",
  LOADING: "LOADING",
  LOADED: "LOADED",
};
