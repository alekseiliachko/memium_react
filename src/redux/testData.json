{
  "authReducer": {
    "username": "alex",
    "authState": "SUCCESS"
  },
  "userReducer": {
    "avatar": "blob:http://localhost:3000/3dcb8dbb-23c7-4abf-8958-52663e0013da",
    "blackList": [
      {
        "accountId": "32df86b7-872d-45d5-8a51-e217f0bc973d",
        "username": "bob",
        "name": null,
        "bio": "populated",
        "imageData": null
      }
    ],
    "details": {
      "accountId": "5d48bb9e-f1d9-41d2-bb02-9c4f578bd282",
      "name": "Круто чел",
      "bio": "populated",
      "gender": "gender",
      "dob": 1615046987793
    },
    "subs": [
      {
        "accountId": "22bb3fe5-185c-4c87-8633-1586586506b7",
        "username": "zilber",
        "name": null,
        "bio": "populated",
        "imageData": null
      }
    ],
    "likedList": [
      {
        "id": "9064e587-3c29-4547-a5a6-51a8149456f8",
        "imageUrl": null,
        "authorId": "5d48bb9e-f1d9-41d2-bb02-9c4f578bd282",
        "title": "Русские мемчанские",
        "description": null,
        "date": 1615046988090,
        "category": "Anime"
      },
      {
        "id": "d3268ef9-bb52-4967-b1ee-14826b9d8a18",
        "imageUrl": null,
        "authorId": "22bb3fe5-185c-4c87-8633-1586586506b7",
        "title": "Chem?",
        "description": null,
        "date": 1615046988090,
        "category": "Chemistry"
      },
      {
        "id": "5d8e2ff1-21bc-4fe0-936f-851407318748",
        "imageUrl": null,
        "authorId": "32df86b7-872d-45d5-8a51-e217f0bc973d",
        "title": "It?",
        "description": null,
        "date": 1615046988090,
        "category": "IT"
      },
      {
        "id": "5b28e638-ef12-45a7-b629-84350cc0c92e",
        "imageUrl": null,
        "authorId": "5d48bb9e-f1d9-41d2-bb02-9c4f578bd282",
        "title": "Bio?",
        "description": null,
        "date": 1615047293904,
        "category": "IT"
      }
    ],
    "avatarLoadingStatus": "LOADED",
    "detailsLoadingStatus": "LOADED",
    "subsLoadingStatus": "LOADED",
    "blackListLoadingStatus": "LOADED",
    "likedListLoadingStatus": "LOADED",
    "feedLoadingStatus": "LOADING",
    "feed": {
      "articles": [],
      "accounts": null
    },
    "myArticles": [
      {
        "id": "9064e587-3c29-4547-a5a6-51a8149456f8",
        "imageUrl": null,
        "authorId": "5d48bb9e-f1d9-41d2-bb02-9c4f578bd282",
        "title": "Русские мемчанские",
        "description": null,
        "date": 1615046988090,
        "category": "Anime"
      },
      {
        "id": "5b28e638-ef12-45a7-b629-84350cc0c92e",
        "imageUrl": null,
        "authorId": "5d48bb9e-f1d9-41d2-bb02-9c4f578bd282",
        "title": "Bio?",
        "description": null,
        "date": 1615047293904,
        "category": "IT"
      }
    ],
    "myArticlesLoadingStatus": "LOADED"
  },
  "articleReducer": {
    "articleDataById": {
      "d3268ef9-bb52-4967-b1ee-14826b9d8a18": {
        "id": "d3268ef9-bb52-4967-b1ee-14826b9d8a18",
        "imageUrl": null,
        "description": null,
        "authorId": "22bb3fe5-185c-4c87-8633-1586586506b7",
        "title": "Chem?",
        "date": 1615046988090,
        "category": "Chemistry",
        "data": "yes, and Jews are responsible."
      },
      "5d8e2ff1-21bc-4fe0-936f-851407318748": {
        "id": "5d8e2ff1-21bc-4fe0-936f-851407318748",
        "imageUrl": null,
        "description": null,
        "authorId": "32df86b7-872d-45d5-8a51-e217f0bc973d",
        "title": "It?",
        "date": 1615046988090,
        "category": "IT",
        "data": "yes, and Jews are responsible."
      },
      "9064e587-3c29-4547-a5a6-51a8149456f8": {
        "id": "9064e587-3c29-4547-a5a6-51a8149456f8",
        "imageUrl": null,
        "description": null,
        "authorId": "5d48bb9e-f1d9-41d2-bb02-9c4f578bd282",
        "title": "Русские мемчанские",
        "date": 1615046988090,
        "category": "Anime",
        "data": "yes, and Jews are responsible."
      },
      "5b28e638-ef12-45a7-b629-84350cc0c92e": {
        "id": "5b28e638-ef12-45a7-b629-84350cc0c92e",
        "imageUrl": null,
        "description": null,
        "authorId": "5d48bb9e-f1d9-41d2-bb02-9c4f578bd282",
        "title": "Bio?",
        "date": 1615047293904,
        "category": "IT",
        "data": "<div><p>Рассмотрим две <b>штуки</b></p><p>Первое:</p><p><b><br></b></p></div>"
      }
    },
    "articleIdsByAuthorName": {}
  },
  "allUsersReducer": {
    "userDataById": {
      "22bb3fe5-185c-4c87-8633-1586586506b7": {
        "accountId": "22bb3fe5-185c-4c87-8633-1586586506b7",
        "name": null,
        "bio": "populated",
        "gender": "gender",
        "dob": 1615046987793,
        "avatar": "blob:http://localhost:3000/595ee9e0-fbd4-4712-98e1-1fd759dca517"
      },
      "32df86b7-872d-45d5-8a51-e217f0bc973d": {
        "accountId": "32df86b7-872d-45d5-8a51-e217f0bc973d",
        "name": null,
        "bio": "populated",
        "gender": "gender",
        "dob": 1615046987793,
        "avatar": "blob:http://localhost:3000/2bd315a7-60e0-41b7-b856-e48559d16d73"
      },
      "5d48bb9e-f1d9-41d2-bb02-9c4f578bd282": {
        "accountId": "5d48bb9e-f1d9-41d2-bb02-9c4f578bd282",
        "name": "Круто чел",
        "bio": "populated",
        "gender": "gender",
        "dob": 1615046987793,
        "avatar": "blob:http://localhost:3000/678d4330-3d93-4470-98f3-4619ed42c932"
      }
    }
  },
  "searchReducer": {
    "articles": [],
    "authors": []
  }
}
