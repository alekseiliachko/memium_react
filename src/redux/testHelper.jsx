import {Provider} from "react-redux";
import thunk from 'redux-thunk';
import configureMockStore from 'redux-mock-store';
import React from "react";

export const getStubStoreData =  () => require('./testData.json');

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

export const withStore = (initialStore) => {
    const store = mockStore(initialStore || getStubStoreData());
    return function Wrapper({ children }) {
        return <Provider store={store}>
            {children}
        </Provider>
    }
}
