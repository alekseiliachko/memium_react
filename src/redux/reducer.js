import { combineReducers, createStore, applyMiddleware, compose } from "redux";
import thunk from "redux-thunk";
import { authReducer } from "./auth/reducer";
import { userReducer } from "./user/reducer";
import { articleReducer } from "./article/reducer";
import { allUsersReducer } from "./allUsers/reducer";
import { searchReducer } from "./search/reducer";

const rootState = combineReducers({
  authReducer,
  userReducer,
  articleReducer,
  allUsersReducer,
  searchReducer,
});

const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;
const middleware = [thunk];
export function configureStore() {
  const store = createStore(
    rootState,
    composeEnhancers(applyMiddleware(...middleware))
  );
  return store;
}
