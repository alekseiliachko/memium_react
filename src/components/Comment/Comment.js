import React, { useCallback, useEffect, useMemo, useState } from "react";
import {
  Box,
  Typography,
  Avatar,
  Divider,
  Toolbar,
  TextField,
  Button,
  IconButton,
} from "@material-ui/core";
import { makeStyles } from "@material-ui/core/styles";
import { useDispatch, useSelector } from "react-redux";
import { fetchUserAvatar, fetchUserData } from "../../redux/allUsers/actions";
import { ProgressDecorator } from "../ProgressDecorator";
import FavoriteIcon from "@material-ui/icons/Favorite";
import FavoriteBorderIcon from "@material-ui/icons/FavoriteBorder";
import BlockIcon from "@material-ui/icons/Block";
import ChatIcon from "@material-ui/icons/Chat";
import { LOADING_STATUS } from "../../redux/user/reducer";
import { loadLikedList } from "../../redux/user/actions";
import AccountsController from "../../api/AccountsController";
import { useHistory } from "react-router-dom";
import CommentsController from "../../api/CommentsController";
import { useLazyUserData } from "../../redux/user/hooks";
import ClearIcon from "@material-ui/icons/Clear";

const useStyles = makeStyles(() => ({
  wrap: {
    display: "flex",
    marginBottom: "20px",
    position: "relative",
  },
  deleteIcon: {
    cursor: "pointer",
    position: "absolute",
    right: "0",
    top: "0",
  },
  textBlock: {
    marginLeft: "10px",
  },
}));

export const Comment = ({ comment, isOwn, deleteComment }) => {
  const classes = useStyles();
  const authorData = useLazyUserData(comment.authorId);

  const dateComp = useMemo(() => {
    const mo = new Intl.DateTimeFormat("ru", { month: "short" }).format(
      comment.date
    );
    const da = new Intl.DateTimeFormat("ru", { day: "2-digit" }).format(
      comment.date
    );

    return (
      <Typography component="span" variant="subtitle2">
        {`, ${mo} ${da}`}
      </Typography>
    );
  }, [comment.date]);

  const handleDeleteComment = useCallback(() => {
    deleteComment(comment.id);
  }, [comment.id]);

  if (!authorData) {
    return null;
  }

  return (
    <div className={classes.wrap}>
      <Avatar src={authorData.avatar} />
      <div className={classes.textBlock}>
        {isOwn && (
          <div className={classes.deleteIcon} onClick={handleDeleteComment}>
            <ClearIcon />
          </div>
        )}
        <Typography component="span" variant="subtitle2">
          {authorData.name}
        </Typography>
        {dateComp}
        <div>{comment.content}</div>
      </div>
    </div>
  );
};
