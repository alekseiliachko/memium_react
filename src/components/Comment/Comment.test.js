import React from "react";
import renderer from "react-test-renderer";

import { Comment } from "./Comment";
import { withStore } from "../../redux/testHelper";

it("renders correctly", async () => {
  let tree;

  const Comp = withStore();
  const deleteCommentMock = jest.fn();
  await renderer.act(async () => {
    tree = renderer.create(
      <Comp>
        <Comment
          deleteComment={deleteCommentMock}
          isOwn={true}
          comment={{
            id: "a4d3c5f2-a57d-42f6-8496-d192066855b4",
            authorId: "5d48bb9e-f1d9-41d2-bb02-9c4f578bd282",
            articleId: "5b28e638-ef12-45a7-b629-84350cc0c92e",
            content: "123",
            date: 1615047212491,
          }}
        />
      </Comp>
    );
    await Promise.resolve();
  });

  expect(tree.toJSON()).toMatchSnapshot();
});
