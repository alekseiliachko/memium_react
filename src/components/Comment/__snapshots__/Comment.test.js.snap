// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="makeStyles-wrap-1"
>
  <div
    className="MuiAvatar-root MuiAvatar-circle"
  >
    <img
      className="MuiAvatar-img"
      src="blob:http://localhost:3000/678d4330-3d93-4470-98f3-4619ed42c932"
    />
  </div>
  <div
    className="makeStyles-textBlock-3"
  >
    <div
      className="makeStyles-deleteIcon-2"
      onClick={[Function]}
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
        />
      </svg>
    </div>
    <span
      className="MuiTypography-root MuiTypography-subtitle2"
    >
      Круто чел
    </span>
    <span
      className="MuiTypography-root MuiTypography-subtitle2"
    >
      , Mar 06
    </span>
    <div>
      123
    </div>
  </div>
</div>
`;
